src/views/EventPage.vue

```diff
@@ -0,0 +1,250 @@
+              :to="`/company/${event.get('company').id}`"
+            >
+              <li>Client Company: {{ companyName }}</li>
+            </router-link>
+          </ul>
+        </section>
+        <section class="fields">
+          <div class="notesTitle">
+            <h2>Notes</h2>
+            <button
+              class="primary"
+              v-if="!editingNotes"
+              @click="editingNotes = true"
+              style="margin-left: 10px; margin-bottom: 10px;"
+            >
+              Edit Notes
+            </button>
+          </div>
+          <div style="margin-top: 5px;">
+            <div v-if="!editingNotes" style="display: flex;">
+              <vue-markdown v-if="!editingNotes">{{
+                event.get("notes")
+              }}</vue-markdown>
+            </div>
+            <div v-if="editingNotes">
+              <textarea v-model="pendingChanges.notes"> </textarea>
+              <button class="primary" @click="saveNote(event)">
+                Save Note
+              </button>
+            </div>
+          </div>
+        </section>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import AV from "leancloud-storage";
+import VueMarkdown from "vue-markdown";
+export default {
+  name: "EventPage",
+  components: {
+    VueMarkdown
+  },
+  data() {
+    return {
+      event: new AV.Object("Event"),
+      clientName: "",
+      companyName: "",
+      editingTime: false,
+      editingNotes: false,
+      pendingChanges: {
+        date: "",
+        time: "",
+        notes: ""
+      }
+    };
+  },
+  methods: {
+    getClientName(id) {
+      const vm = this;
+      const clientQuery = new AV.Query("Client");
+      clientQuery
+        .get(id)
+        .then(client => {
+          vm.clientName = client.get("fullName");
+          vm.getCompanyName(client.get("company").get("objectId"));
+        })
+        .catch(error => {
+          alert(error);
+        });
+    },
+    getCompanyName(id) {
+      const vm = this;
+      const companyQuery = new AV.Query("Company");
+      companyQuery
+        .get(id)
+        .then(company => {
+          vm.companyName = company.get("name");
+        })
+        .catch(error => {
+          alert(error);
+        });
+    },
+    toggle(event) {
+      const vm = this;
+      event
+        .set("done", !event.get("done"))
+        .save()
+        .then(vm.created)
+        .catch(error => {
+          alert(error);
+        });
+    },
+    update(event) {
+      const vm = this;
+      event.set(
+        "time",
+        new Date(
+          vm.pendingChanges.date.slice(0, 4),
+          vm.pendingChanges.date.slice(5, 7) - 1,
+          vm.pendingChanges.date.slice(8, 10),
+          vm.pendingChanges.time.slice(0, 2),
+          vm.pendingChanges.time.slice(3, 5),
+          0
+        )
+      );
+      event
+        .save()
+        .then(() => {
+          vm.editingTime = false;
+          vm.created;
```

Comment:
What’s this line for?

Reply:
wasn't sure if I needed to call the created function again to load the data.